hazelcast/src/main/java/com/hazelcast/sql/impl/exec/scan/index/IndexInFilter.java

```diff
@@ -168,7 +168,7 @@ private LazyIterator(InternalIndex index, ExpressionEvalContext evalContext, Col
             while (filterIterator.hasNext()) {
                 IndexFilter filter = filterIterator.next();
 
-                Iterator<QueryableEntry> iterator = filter.getEntries(index, evalContext);
+                Iterator<QueryableEntry> iterator = filter.getEntries(index, false, evalContext);
```

Comment:
Why do we pass `false` here? Wouldn't it break the expected collation?

Reply:
Fixed in the https://github.com/hazelcast/hazelcast/pull/18013/commits/5abe68ec3c2906a47c2cfce80b69aac0dc483e43